src/LandIce/evaluators/LandIce_BasalFrictionCoefficient_Def.hpp

```diff
@@ -242,115 +303,130 @@ KOKKOS_INLINE_FUNCTION
+        } else
+          Nval = N(cell,ipt);
+        break;
+      case EFFECTIVE_PRESSURE_TYPE::ONE:
+        if(zero_N_on_floating_at_nodes) {
+          Nval = 0;
+          if(nodal) {
+            if(rho_i*thickness_field(cell,ipt)+rho_w*bed_topo_field(cell,ipt) > 0)
+              Nval =  1.0;
+          } else {
+            for (unsigned int node=0; node<numNodes; ++node)
+              if (rho_i*thickness_field(cell,node)+rho_w*bed_topo_field(cell,node) > 0)
+                Nval += BF(cell,node,ipt);
           }
         }
-      } else {
-        if (distributedMu) {
-          for (unsigned int ipt=0; ipt<dim; ++ipt) {
-            ScalarT Nval = N(cell,ipt);
-            if (is_side_equation) Nval = KU::max(Nval, 0.0);
-            ScalarT q = u_norm(cell,ipt) / ( u_norm(cell,ipt) + lambda*ice_softness(cell)*std::pow(Nval,n) );
-            beta(cell,ipt) = muCoulombField(cell,ipt) * Nval * std::pow( q, power) / u_norm(cell,ipt);
-          }
-        } else {
-          for (unsigned int ipt=0; ipt<dim; ++ipt) {
-            ScalarT Nval = N(cell,ipt);
-            if (is_side_equation) Nval = KU::max(Nval, 0.0);
-            ScalarT q = u_norm(cell,ipt) / ( u_norm(cell,ipt) + lambda*ice_softness(cell)*std::pow(Nval,n) );
-            beta(cell,ipt) = mu * Nval * std::pow( q, power) / u_norm(cell,ipt);
+        break;
+      case EFFECTIVE_PRESSURE_TYPE::HYDROSTATIC:
+        if(nodal)
+          Nval = g*KU::max(rho_i*thickness_field(cell,ipt)+KU::min(rho_w*bed_topo_field(cell,ipt),0.0),0.0);
+        else {
+          MeshScalarT thickness(0), bed_topo(0);
+          for (unsigned int node=0; node<numNodes; ++node) {
+            thickness += thickness_field(cell,node)*BF(cell,node,ipt);
+            bed_topo += bed_topo_field(cell,node)*BF(cell,node,ipt);
           }
+          Nval = g*KU::max(rho_i*thickness+KU::min(rho_w*bed_topo,0.0),0.0);
+        }
+        break;
+      case EFFECTIVE_PRESSURE_TYPE::HYDROSTATIC_AT_NODES:
+        if(nodal)
+          Nval = g* KU::max(rho_i*thickness_field(cell,ipt)+KU::min(rho_w*bed_topo_field(cell,ipt),0.0),0.0);
+        else {
+          Nval = 0;
+          for (unsigned int node=0; node<numNodes; ++node)
+            Nval += g*KU::max(rho_i*thickness_field(cell,node)+KU::min(rho_w*bed_topo_field(cell,node),0.0),0.0)*BF(cell,node,ipt);
         }
+        break;
       }
-    break;
-
-    case EXP_GIVEN_FIELD:
-      if(nodal || interpolate_then_exponentiate) {
-        if (is_given_field_param) {
-          for (unsigned int ipt=0; ipt<dim; ++ipt)
-            beta(cell,ipt) = std::exp(given_field_param(cell,ipt));
-        } else {
-          for (unsigned int ipt=0; ipt<dim; ++ipt)
-            beta(cell,ipt) = std::exp(given_field(cell,ipt));
+
+      switch (mu_type) {
+      case FIELD_TYPE::FIELD:
+        muValue = muField(cell,ipt);
+        break;
+      case FIELD_TYPE::EXPONENT_OF_FIELD_AT_NODES:
```

Comment:
Is this option really something we want? I think it's easier (and probably more accurate) to interpolate mu at the quadrature points, and then exponentiate, no?

Reply:
Ah, wait: is the reasoning that mu might be the log of the field, and you would like to get the interpolation of the actual nodal mu, rather than `exp(interp(log(mu)))` ? If that's the case, then I would get rid of EXPONENT_OF_FIELD. I mean, what's the use of it?

Reply:
That's exactly right. And I'd like to get rid of EXPONENT OF FIELD, just wanted to keep backword compatibility for now, as that's the one we use in most of the tests